.evergreen/config.yml

```diff
@@ -0,0 +1,262 @@
+#                Tasks                #
+#######################################
+tasks:
+  - name: "static-analysis-task"
+    tags:
+      - "pr"
+    commands:
+      - command: "subprocess.exec"
+        type: "test"
+        params:
+          working_dir: "src"
+          binary: "bash"
+          add_to_path:
+            - ".evergreen"
+          args:
+            - "static-checks.sh"
+
+  - name: "unit-tests-task"
+    tags:
+      - "pr"
+    commands:
+      - command: "subprocess.exec"
+        type: "test"
+        params:
+          working_dir: "src"
+          binary: "bash"
+          add_to_path:
+            - ".evergreen"
+          args:
+            - "run-unit-tests.sh"
+
+  - name: "integration-tests-task"
+    tags:
+      - "pr"
+    commands:
+      - func: "bootstrap-mongo-orchestration"
+      - command: "subprocess.exec"
+        type: "test"
+        params:
+          working_dir: "src"
+          binary: "bash"
+          add_to_path:
+            - ".evergreen"
+          args:
+            - "run-integration-tests.sh"
+
+  - name: "publish-snapshots-task"
+    depends_on:
+      - variant: "static-checks"
+        name: "static-analysis-task"
+    commands:
+      - command: "subprocess.exec"
+        params:
+          working_dir: "src"
+          binary: "bash"
+          add_to_path:
+            - ".evergreen"
+          env:
+            NEXUS_USERNAME: "${nexus_username}"
+            NEXUS_PASSWORD: "${nexus_password}"
+            SIGNING_PASSWORD: "${signing_password}"
+            SIGNING_KEY: "${signing_key}"
+            RELEASE: "false"
+          args:
+            - ".evergreen/publish.sh"
+
+  - name: "publish-release-task"
+    git_tag_only: true
+    commands:
+      - command: "subprocess.exec"
+        params:
+          working_dir: "src"
+          binary: "bash"
+          add_to_path:
+            - ".evergreen"
+          env:
+            NEXUS_USERNAME: "${nexus_username}"
+            NEXUS_PASSWORD: "${nexus_password}"
+            SIGNING_PASSWORD: "${signing_password}"
+            SIGNING_KEY: "${signing_key}"
+            RELEASE: "true"
+          args:
+            - ".evergreen/publish.sh"
+
+########################################
+#              Axes                    #
+########################################
+axes:
+  - id: "mongo-version"
+    display_name: "MongoDB Version"
+    values:
+      - id: "8.0"
+        display_name: "8.0"
+        variables:
+          MONGODB_VERSION: "8.0"
+      - id: "7.0"
+        display_name: "7.0"
+        variables:
+          MONGODB_VERSION: "7.0"
+      - id: "6.0"
+        display_name: "6.0"
+        variables:
+          MONGODB_VERSION: "6.0"
+
+  - id: "topology"
+    display_name: "Topology"
+    values:
+      - id: "sharded"
+        display_name: "Sharded Cluster"
+        variables:
+          TOPOLOGY: "sharded_cluster"
+
+  - id: "os"
+    display_name: "OS"
+    values:
+      - id: "linux"
```

Comment:
do we usually only test on linux 
Looking at the documentation looks like we support a wide range of different OS's
https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files#matrix-variant-definition